### DISCUSSION TITLE
[RAC] How to ensure one modal is open at a time?

### DISCUSSION THREAD

**Discussion by devlzcode** [adobe/react-spectrum#6389] [Q&A]
I have a mobile bottom bar that has two buttons that open modals. The modal is placed right above the bottom bar and the underlay is beneath them both, meaning that the bottom bar is interactable while the modal is open, this is intended because it's how I want the modals to be closed by pressing their respective buttons again. The issue is that if you press one trigger then another, you can have two open at once. How can I fix this? Are modals the right component for this, or should I make a custom one with react-aria (although using it is confusing for me at the moment)?

<img width="427" alt="image" src="https://github.com/adobe/react-spectrum/assets/58056533/35a8193d-1b21-4a72-ad1b-2868b668fd5b">

<img width="434" alt="image" src="https://github.com/adobe/react-spectrum/assets/58056533/00002521-0b19-4348-b8bf-5d747232778c">

**Comment by snowystinger** [comment]
You're going to have issues with focus management if the toolbar is not either part of the modal or behind the underlay.

Could you add a copy of the toolbar to the tray?

Another option could be to render two elements for the underlay, one that is semi transparent, the other, completely transparent. That way you can still have the underlay over the toolbar, it just wouldn't look like it.
